test/field/Steppers.test.cu

```diff
@@ -79,57 +80,83 @@ __global__ void rk4_test_kernel(FieldTestParams param,
-__global__ void dp547_test_kernel(FieldTestParams param,
+// KERNELS
+//---------------------------------------------------------------------------//
+__global__ void helix_test_kernel(FieldTestParams param,
                                   real_type*      pos_x,
                                   real_type*      pos_z,
                                   real_type*      mom_y,
                                   real_type*      mom_z,
                                   real_type*      error)
 {
-    auto tid = celeritas::KernelParamCalculator::thread_id();
-    if (tid.get() >= param.nstates)
-        return;
+    UniformZMagField field(param.field_value);
+
+    gpu_stepper<UniformZMagField, ZHelixStepper>(
+        param, field, pos_x, pos_z, mom_y, mom_z, error);
+}
 
-    // Construct the DormandPrinceStepper stepper
+//---------------------------------------------------------------------------//
+__global__ void rk4_test_kernel(FieldTestParams param,
+                                real_type*      pos_x,
+                                real_type*      pos_z,
+                                real_type*      mom_y,
+                                real_type*      mom_z,
+                                real_type*      error)
+{
     UniformMagField field({0, 0, param.field_value});
-    using RKTraits
-        = detail::MagTestTraits<UniformMagField, DormandPrinceStepper>;
-    RKTraits::Equation_t equation(field, units::ElementaryCharge{-1});
-    RKTraits::Stepper_t  dp547(equation);
 
-    // Initial state and the epected state after revolutions
-    OdeState y;
-    y.pos = {param.radius, 0.0, tid.get() * 1.0e-6};
-    y.mom = {0.0, param.momentum_y, param.momentum_z};
+    gpu_stepper<UniformMagField, RungeKuttaStepper>(
+        param, field, pos_x, pos_z, mom_y, mom_z, error);
+}
 
-    // Test parameters and the sub-step size
-    real_type hstep       = 2.0 * constants::pi * param.radius / param.nsteps;
-    real_type total_error = 0;
+//---------------------------------------------------------------------------//
+__global__ void dp547_test_kernel(FieldTestParams param,
+                                  real_type*      pos_x,
+                                  real_type*      pos_z,
+                                  real_type*      mom_y,
+                                  real_type*      mom_z,
+                                  real_type*      error)
+{
+    UniformMagField field({0, 0, param.field_value});
 
-    for (auto nr : range(param.revolutions))
-    {
-        // Travel hstep for nsteps times in the field
-        for (CELER_MAYBE_UNUSED int i : celeritas::range(param.nsteps))
-        {
-            StepperResult result = dp547(hstep, y);
-            y                    = result.end_state;
-            total_error += truncation_error(hstep, 0.001, y, result.err_state);
-        }
-    }
+    gpu_stepper<UniformMagField, DormandPrinceStepper>(
```

Comment:
Just to point out here, since your test is only testing the steppers and not the magnetic field classes, you could just `UniformZMagField` for all these test cases.